src/actions/LoginThunkActions.ts

```diff
@@ -1,34 +1,69 @@
+import { KialiAppState, Token, ServerConfig } from '../store/Store';
 import { KialiAppAction } from './KialiAppAction';
 import HelpDropdownThunkActions from './HelpDropdownThunkActions';
 import GrafanaThunkActions from './GrafanaThunkActions';
 import { LoginActions } from './LoginActions';
 import * as API from '../services/Api';
-import { ServerConfig } from '../config/Config';
+import { ServerConfigActions } from './ServerConfigActions';
 
+const ANONYMOUS: string = 'anonymous';
+
+// completeLogin performs any initialization required prior to declaring the login complete. This
+// prevents early rendering for components waiting on a login.
+const completeLogin = (
+  dispatch: ThunkDispatch<KialiAppState, void, KialiAppAction>,
+  token: Token,
+  username: string,
+  timeout?: number
+) => {
+  const auth = `Bearer ${token['token']}`;
+  API.getServerConfig(auth).then(
+    response => {
+      // set the serverConfig before completing login so that it is available for
+      // anything needing it to render properly.
+      const serverConfig: ServerConfig = {
+        istioNamespace: response.data.istioNamespace,
+        istioLabels: response.data.istioLabels,
+        prometheus: response.data.prometheus
+      };
+      dispatch(ServerConfigActions.setServerConfig(serverConfig));
+
+      // complete the login process
+      dispatch(LoginActions.loginSuccess(token, username, timeout));
+
+      // dispatch requests to be done now but not necessarily requiring immediate completion
+      dispatch(HelpDropdownThunkActions.refresh());
+      dispatch(GrafanaThunkActions.getInfo(auth));
+    },
+    error => {
+      /** Logout user */
+      if (error.response && error.response.status === HTTP_CODES.UNAUTHORIZED) {
+        dispatch(LoginActions.logoutSuccess());
+      }
+    }
+  );
+};
+
+// performLogin performs only the authentication part of login. If successful
+// we call completeLogin to perform any initialization required for the user session.
 const performLogin = (
   dispatch: ThunkDispatch<KialiAppState, void, KialiAppAction>,
   username?: string,
   password?: string
 ) => {
   dispatch(LoginActions.loginRequest());
 
-  const anonymous = username === undefined;
-  const loginUser: string = username === undefined ? 'anonymous' : username;
-  const loginPass: string = password === undefined ? 'anonymous' : password;
+  const loginUser: string = username === undefined ? ANONYMOUS : username;
+  const loginPass: string = password === undefined ? ANONYMOUS : password;
 
   API.login(loginUser, loginPass).then(
     token => {
-      dispatch(LoginActions.loginSuccess(token['data'], loginUser));
-      const auth = `Bearer ${token['data']['token']}`;
-      dispatch(HelpDropdownThunkActions.refresh());
-      dispatch(GrafanaThunkActions.getInfo(auth));
+      completeLogin(dispatch, token['data'], loginUser);
     },
     error => {
-      if (anonymous) {
+      if (loginUser === ANONYMOUS) {
```

Comment:
better to check `username === undefined`, no? If the user is really named "anonymous" ...

Reply:
@jotak Thanks for proactively taking a look at this PR.  Above we set loginUser to ANONYMOUS based on `username === undefined` and ANONYMOUS has more semantic meaning here than undefined.  So, I think it's OK.

Reply:
Did you actually test anonymous access? I wrote that and IIRC that "undefined" was special. I'll test it today, but want to ask if you already did. I have a sneaky suspicion that anonymous access has been broken.

Reply:
quick smoke test appears to show me that, yes, anonymous access has now been broken. We need to fix this. We know of at least one community member using anonymous access.

Reply:
This most likely was broke in another PR where I changed it to support the situation when there is no secret installed. I'll write up a JIRA.